encode.go

```diff
@@ -102,13 +102,13 @@ func (e *encoder) marshalDoc(tag string, in reflect.Value) {
 	} else {
 		yaml_document_start_event_initialize(&e.event, nil, nil, true)
 		e.emit()
-		e.marshal(tag, in)
+		e.marshal(tag, in, nil, nil, nil)
```

Comment:
Maybe we can have a marshallWithMetadata function to that marshall wraps so that we can reduce the number of line changes?

```golang
func (e *encoder) marshal(tag string, in reflect.Value) {
  e.marsallWithMetadata(tag, in, nil, nil, nil)
}
```

in general is there a way to clearly separate the parts of the code that use metadata and the parts that don't?

Reply:
Unfortunately not, its all recursive calls through `marshal` for the various arrays/structs/maps until it hits the bottom and uses `stringv` `boolv` `intv` `uintv` `floatv` or `nilv` to convert the primitve to string and call `emitScalar`.

To use metaData I need comments to exist on the recursive function, so that they can get passed down to the correct primitive call `stringv` `intv` etc. So it would end up duplicating essentially the entire function with 2 seperate recursive paths where the first can cutover to the second at any point.